src/ray/common/ray_object.cc

```diff
@@ -14,6 +14,62 @@
 
 #include "ray/common/ray_object.h"
 
+#include "msgpack.hpp"
+
+namespace {
+
+/// Serialize the protobuf message to msg pack.
+///
+/// Ray uses Msgpack for cross-language object serialization.
+/// This method creates a msgpack serialized buffer that contains
+/// serialized protobuf message.
+///
+/// Language frontend can deseiralize this object to obtain
+/// data stored in a given protobuf. Check `serialization.py` to see
+/// how this works.
+///
+/// NOTE: The function guarantees that the returned buffer contains data.
+///
+/// \param protobuf_message The protobuf message to serialize.
+/// \return The buffer that contains serialized msgpack message.
+template <class ProtobufMessage>
+std::unique_ptr<ray::LocalMemoryBuffer> SerializeErrorInfo(
+    const ProtobufMessage &protobuf_message) {
+  // Structure of bytes stored in object store:
+
+  // First serialize RayException by the following steps:
+  // PB's RayException
+  // --(PB Serialization)-->
+  // --(msgpack Serialization)-->
```

Comment:
NOTE: This serialization format is a bit weird. We technically don't need to have msgpack serialization here (only need to serialize protobuf). I asked the original author why it was done in this way, and he said there was no special reason (he just followed other parts of code).

That says, we want to move onto removing msgpack here and just directly store serialized protobuf. This is TODO

Reply:
yeah; it's weird.